### ISSUE TITLE
ENOPKG Missing `package.json` file.

### ISSUE THREAD

**Issue by estahn** [semantic-release/semantic-release#738]
Using `semantic-release` with non-npm/nodejs projects throws an error:

ENOPKG Missing `package.json` file.

Is this project only for nodejs projects?

**Comment by pvdlg** [comment]
No it's not only for Node.

Could you provide the information requested in the issue report template? Otherwise there is nothing we can do to help with an error code.

**Comment by estahn** [comment]
I wanted to test if I made a mistake and created a fresh directory with the following outcome:

```                                                                                                                                                                     
 …/p/hipages  mkdir test-semantic-release                                                                                                                                                                            …/p/hipages  cd test-semantic-release                                                                                                                                                                               …/hipages/test-semantic-release  git init                                                                                                                                                                          Initialized empty Git repository in /Users/estahn/p/hipages/test-semantic-release/.git/
 …/hipages/test-semantic-release    master  semantic-release --dry-run                                                                                                                                            [Semantic release]: Running semantic-release version 15.1.5
[Semantic release]: Load plugin verifyConditions from @semantic-release/npm
[Semantic release]: Load plugin verifyConditions from @semantic-release/github
[Semantic release]: Load plugin analyzeCommits from @semantic-release/commit-analyzer
[Semantic release]: Load plugin generateNotes from @semantic-release/release-notes-generator
[Semantic release]: Load plugin prepare from @semantic-release/npm
[Semantic release]: Load plugin publish from @semantic-release/npm
[Semantic release]: Load plugin publish from @semantic-release/github
[Semantic release]: Load plugin success from @semantic-release/github
[Semantic release]: Load plugin fail from @semantic-release/github
[Semantic release]: This test run was triggered on the branch undefined, while semantic-release is configured to only publish from master, therefore a new version won’t be published.
 …/hipages/test-semantic-release    master  semantic-release --version                                                                                                                                             ✔  luke-admin/default ⎈  3602  16:47:40
15.1.5
 …/hipages/test-semantic-release    master                                                                                                                                                                         ✔  luke-admin/default ⎈  3603  16:47:53
```

Next, running it on a repository:
```
git clone https://github.com/hipages/php-fpm_exporter
cd php-fpm_exporter
semantic-release --dry-run                                                                                                                                 ✔  luke-admin/default ⎈  3609  16:50:37
[Semantic release]: Running semantic-release version 15.1.5
[Semantic release]: Load plugin verifyConditions from @semantic-release/npm
[Semantic release]: Load plugin verifyConditions from @semantic-release/github
[Semantic release]: Load plugin analyzeCommits from @semantic-release/commit-analyzer
[Semantic release]: Load plugin generateNotes from @semantic-release/release-notes-generator
[Semantic release]: Load plugin prepare from @semantic-release/npm
[Semantic release]: Load plugin publish from @semantic-release/npm
[Semantic release]: Load plugin publish from @semantic-release/github
[Semantic release]: Load plugin success from @semantic-release/github
[Semantic release]: Load plugin fail from @semantic-release/github
[Semantic release]: Run automated release from branch master
[Semantic release]: Call plugin verify-conditions
[Semantic release]: Verify GitHub authentication
[Semantic release]: ENOPKG Missing `package.json` file.
A package.json file (https://docs.npmjs.com/files/package.json) at the root of your project is required to release on npm.

Please follow the npm guideline (https://docs.npmjs.com/getting-started/creating-node-modules) to create a valid package.json file.


{ SemanticReleaseError: Missing `package.json` file.
    at module.exports (/usr/local/lib/node_modules/semantic-release/node_modules/@semantic-release/npm/lib/get-error.js:6:10)
    at module.exports (/usr/local/lib/node_modules/semantic-release/node_modules/@semantic-release/npm/lib/get-pkg.js:17:19)
    at <anonymous>
  name: 'SemanticReleaseError',
  code: 'ENOPKG',
  details: 'A [package.json file](https://docs.npmjs.com/files/package.json) at the root of your project is required to release on npm.\n\nPlease follow the [npm guideline](https://docs.npmjs.com/getting-started/creating-node-modules) to create a valid `package.json` file.',
  semanticRelease: true,
  pluginName: '@semantic-release/npm' }```

**Comment by pvdlg** [comment]
As indicated in the [plugin configuration doc](https://semantic-release.gitbooks.io/semantic-release/content/docs/usage/plugins.html) by default semantic-release use the [@semantic-release/npm](https://github.com/semantic-release/npm) plugin for `verifyConditions`, `prepare` and `publish` steps, which update the `package.json` and publish on the npm registry. This plugin require a `package.json`.

If you don't have a `package.json` you should disable the npm plugin and replace it by another plugin to make the releases.
Depending on the project type you have several options:
- Use the [@semantic-release/git](https://github.com/semantic-release/git) plugin to commit compiled assets to your git repo
- Use the [@semantic-release/exec](https://github.com/semantic-release/exec) plugin to run a custom shell command in the `publish` step in order to publish to your package registry
- Create your own plugin to publish to your package registry if the capabilities of the [@semantic-release/exec](https://github.com/semantic-release/exec) plugin are not sufficient

**Comment by pvdlg** [comment]
I updated the FAQ for non-Node project in #743. Hopefully that helps clarifying the steps to follow.

**Comment by estahn** [comment]
Works like a charm! Thanks for your help @pvdlg

**Comment by Shereef** [comment]
tl;dr you may have forgotten to include your `.releaserc`
